src/main/java/com/gmail/nossr50/skills/mining/MiningManager.java

```diff
@@ -31,55 +31,78 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
-import java.util.Set;
 
 import static com.gmail.nossr50.util.ItemUtils.isPickaxe;
 
 public class MiningManager extends SkillManager {
 
-    public static final String BUDDING_AMETHYST = "budding_amethyst";
-    public static final Collection<Material> BLAST_MINING_BLACKLIST = Set.of(Material.SPAWNER);
-
     public MiningManager(@NotNull McMMOPlayer mcMMOPlayer) {
         super(mcMMOPlayer, PrimarySkillType.MINING);
     }
 
+    /**
+     * Determines if the player can use Demolitions Expertise.
+     *
+     * @return true if the player can use Demolitions Expertise, false otherwise
+     */
     public boolean canUseDemolitionsExpertise() {
-        if (!RankUtils.hasUnlockedSubskill(getPlayer(), SubSkillType.MINING_DEMOLITIONS_EXPERTISE))
-            return false;
-
-        return getSkillLevel() >= BlastMining.getDemolitionExpertUnlockLevel() && Permissions.demolitionsExpertise(getPlayer());
+        return RankUtils.hasUnlockedSubskill(getPlayer(), SubSkillType.MINING_DEMOLITIONS_EXPERTISE)
+                && getSkillLevel() >= BlastMining.getDemolitionExpertUnlockLevel()
+                && Permissions.demolitionsExpertise(getPlayer());
     }
 
+    /**
+     * Determines if the player can detonate TNT remotely.
+     *
+     * @return true if the player can detonate TNT remotely, false otherwise
+     */
     public boolean canDetonate() {
         Player player = getPlayer();
-
         return canUseBlastMining() && player.isSneaking()
-                && (isPickaxe(getPlayer().getInventory().getItemInMainHand()) || player.getInventory().getItemInMainHand().getType() == mcMMO.p.getGeneralConfig().getDetonatorItem())
+                && (isPickaxe(player.getInventory().getItemInMainHand())
```

Comment:
I appreciate trying to clean up the mess that is the existing code, but try to keep the diff minimal for review